model/impl.go

```diff
@@ -46,6 +47,175 @@ func newRealCluster(tsConstructor func() store.TimeStore, resolution time.Durati
 	return cluster
 }
 
+// GetClusterMetric returns a metric of the cluster entity, along with the latest timestamp.
+// GetClusterMetric receives as arguments the name of a metric and a starting timestamp.
+// GetClusterMetric returns a slice of TimePoints for that metric, with times starting AFTER the starting timestamp.
+func (rc *realCluster) GetClusterMetric(metric_name string, start time.Time) ([]store.TimePoint, time.Time, error) {
+	rc.lock.RLock()
+	defer rc.lock.RUnlock()
+
+	if len(rc.Metrics) == 0 {
+		return nil, time.Time{}, fmt.Errorf("cluster metrics are not populated yet")
+	}
+
+	ts, ok := rc.Metrics[metric_name]
+	if !ok {
+		return nil, time.Time{}, fmt.Errorf("the requested metric is not present")
+	}
+	res := (*ts).Get(start, time.Time{})
+	return res, rc.timestamp, nil
+}
+
+// GetNodeMetric returns a metric of a node entity, along with the latest timestamp.
+// GetNodeMetric receives as arguments the name of a metric and a starting timestamp.
+// GetNodeMetric returns a slice of TimePoints for that metric, with times starting AFTER the starting timestamp.
+func (rc *realCluster) GetNodeMetric(hostname string, metric_name string, start time.Time) ([]store.TimePoint, time.Time, error) {
+	rc.lock.RLock()
+	defer rc.lock.RUnlock()
+
+	if len(rc.Nodes) == 0 {
+		return nil, time.Time{}, fmt.Errorf("the model is not populated yet")
```

Comment:
Good call. We generally define a zeroTime variable if there are multiple instances of time.Time{} within the same scope